src/js/overlay.js

```diff
@@ -311,6 +312,10 @@ Overlay.prototype.realign = function(dimension, size) {
 		if (!this.opts.arrow) {
 			this.wrapper.style['margin' + utils.capitalise(edge)] = -(this.wrapper['offset' + utils.capitalise(dimension)]/2) + 'px';
 		}
+		// Set via JS as it needs the full size when calculating the negative margins
+		// and setting left: 50% for example before, if the overlay doesn't fit, would
+		// shrink the overlay
```

Comment:
> Set alignment in JavaScript (not via CSS) after all other styles have been applied
so that browsers compute it properly

Not sure that covers it, but hopefully that gives you a head start.